engine/class_modules/paladin/sc_paladin_holy.cpp

```diff
@@ -307,20 +305,22 @@ struct holy_shock_t : public paladin_spell_t
 
   void execute() override
   {
-    paladin_spell_t::execute();
-
     if ( dmg )
     {
       // damage enemy
-      damage->set_target( execute_state->target );
+      damage->set_target( target );
       damage->execute();
     }
     else
     {
       // heal friendly
-      heal->set_target( execute_state->target );
+      heal->set_target( target );
       heal->execute();
     }
+    // The reason the subspell is allowed to execute first is so we can consume judgment correctly. 
+    // If this is first then a dummy holy shock is fired off that consumes the judgment debuff 
+    // meaning when we attempt to do damage judgment will not be factored in even when it should be.
```

Comment:
Might be worth adding that it's consumed because of dumb blizz spelldata that makes every holy shock spell affected by judgment, instead of just affecting the damaging holy shock spell.

Reply:
did done